tests/backup/backup_test.go

```diff
@@ -1951,6 +1953,111 @@ var _ = Describe("{BackupRestoreOverPeriodSimultaneous}", func() {
+		// randomly select 1 character from given charset
+		b[i] = allChar[rand.Intn(len(allChar))]
+	}
+
+	//shuffle character
+	rand.Shuffle(len(b), func(i, j int) {
+		b[i], b[j] = b[j], b[i]
+	})
+
+	return string(b)
+}
+
+// This test cases creates a backup location with encryption
+var _ = Describe("{BackupLocationWithEncryptionKey}", func() {
+	var contexts []*scheduler.Context
+	var appContexts []*scheduler.Context
+	var bkpNamespaces []string
+	var backup_location_name string
+	var CloudCredUID string
+	var cluster_uid string
+	var CredName string
+	var restorename string
+	var cluster_status api.ClusterInfo_StatusInfo_Status
+	providers := getProviders()
+	bucket_name := getBucketName()
+	JustBeforeEach(func() {
+		log.Infof("No pre-setup required for this testcase")
+		StartTorpedoTest("Backup: BackupLocationWithEncryptionKey", "Creating BackupLoaction with Encryption Keys", nil, 0)
+	})
+	It("Creating backup location and cloud setting", func() {
+		log.InfoD("Creating bucket,backup location and cloud setting")
+		CredName = fmt.Sprintf("%s-%s", "cred", providers[0])
+		backup_location_name = fmt.Sprintf("%s-%s", "location", providers[0])
+		CloudCredUID = uuid.New()
+		BackupLocationUID = uuid.New()
+		encryptionkey := generate_encryption_key()
+		CreateCloudCredential(providers[0], CredName, CloudCredUID, orgID)
+		time.Sleep(time.Minute * 1)
+		CreateBackupLocation(providers[0], backup_location_name, BackupLocationUID, CredName, CloudCredUID, bucket_name[0], orgID, encryptionkey)
+		log.InfoD("Deploy applications")
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts = ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = appReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				bkpNamespaces = append(bkpNamespaces, namespace)
+			}
+		}
+
+		Step("Register cluster for backup", func() {
+			cluster_status, cluster_uid = RegisterBackupCluster(orgID, "", "")
+			dash.VerifyFatal(cluster_status, api.ClusterInfo_StatusInfo_Online, "Verifying backup cluster")
+		})
+
+		Step("Taking backup of applications", func() {
+			for _, namespace := range bkpNamespaces {
+				backupName := fmt.Sprintf("%s-%s", BackupNamePrefix, namespace)
+				CreateBackup(backupName, sourceClusterName, backup_location_name, BackupLocationUID, []string{namespace},
+					nil, orgID, cluster_uid, "", "", "", "")
+			}
+		})
+
+		Step("Restoring the backed up application", func() {
+			for _, namespace := range bkpNamespaces {
+				backupName := fmt.Sprintf("%s-%s", BackupNamePrefix, namespace)
+				restorename = fmt.Sprintf("%s-%s", restoreNamePrefix, backupName)
+				CreateRestore(restorename, backupName, nil, destinationClusterName, orgID)
+			}
+		})
+	})
+	JustAfterEach(func() {
+		defer EndTorpedoTest()
+		log.Infof("Cleanup started")
```

Comment:
Do you really need this log? Can we use something like cleaning up Backups and restores?